### PR TITLE
Add operation name as route name for backend route.

### PR DESCRIPTION
**Pull request by shuoyang2016**
Also refactor the listener_genarator code to expose the AddPerRouteConfigGenToMethods.

### CODE REVIEW

File: src/go/configgenerator/listener_generator.go [hunk 1]
```diff
@@ -83,9 +86,18 @@ func MakeListener(serviceInfo *sc.ServiceInfo, filterGenerators []*filterconfig.
 					return nil, err
 				}
 			}
-
 		}
 	}
+	return nil, nil
```
Comment (nareddyt) [inline src/go/configgenerator/listener_generator.go:91]: Return httpFilters?

File: src/go/configgenerator/route_generator.go [hunk 2]
```diff
@@ -403,6 +403,7 @@ func makeRoute(routeMatcher *routepb.RouteMatch, method *configinfo.MethodInfo)
 	}
 
 	return &routepb.Route{
```
Comment (nareddyt) [inline src/go/configgenerator/route_generator.go:405]: There are many other places where a route is constructed in this file. CORS, etc. Do control-f for `&routepb.Route{` just to see a few.

Comment (shuoyang2016) [reply src/go/configgenerator/route_generator.go:405]: Looks like the cors route doesn't associate with an method info? Is that so? In this case, what should be the route name?

**Comment by shuoyang2016** [comment]
Finally fixed all testdata. I didn't anticipate there are so many testdata which took me hours to complete.

File: src/go/configgenerator/listener_generator.go (hunk 1 above)
Comment (shuoyang2016) [reply src/go/configgenerator/listener_generator.go]: Thanks! Done.

**Comment by shuoyang2016** [comment]
PTAL

**Review by qiwzhang — APPROVED** [review]

**Comment by nareddyt** [comment]
> Finally fixed all testdata. I didn't anticipate there are so many testdata which took me hours to complete.

This is exactly why I didn't add unit tests to our config generator in google3 beforehand :) I want to find a better way to do testing internally. Something like this: https://ieftimov.com/post/testing-in-go-golden-files/#keeping-our-golden-files-up-to-date

File: src/go/configgenerator/route_generator.go (hunk 2 above)
Comment (nareddyt) [reply src/go/configgenerator/route_generator.go:405]: Ok, that's fine then.

File: src/go/configgenerator/route_generator_test.go [hunk 3]
```diff
@@ -71,121 +71,123 @@ func TestMakeRouteConfig(t *testing.T) {
 			},
 			wantRouteConfig: `
 {
-    "name": "local_route",
-    "virtualHosts": [
+  "name": "local_route",
```
Comment (nareddyt) [inline src/go/configgenerator/route_generator_test.go:74]: nit: did you change indentation? diff is big

**Review by nareddyt — APPROVED** [review]

File: src/go/configmanager/testdata/test_fixed_mode_dynamic_routing.go [hunk 4]
```diff
@@ -599,6 +608,7 @@ var (
                         ],
                         "path": "/search"
                       },
+											"name": "1.echo_api_endpoints_cloudesf_testing_cloud_goog.dynamic_routing_Search",
```
Comment (TAOXUY) [inline src/go/configmanager/testdata/test_fixed_mode_dynamic_routing.go:611]: NIT, indent

File: src/go/configmanager/testdata/test_fetch_listeners.go [hunk 5]
```diff
@@ -2457,6 +2486,7 @@ var (
                         ],
                         "path": "/simplegetcors"
                       },
+											"name": "1.echo_api_endpoints_cloudesf_testing_cloud_goog.Simplegetcors",
```
Comment (TAOXUY) [inline src/go/configmanager/testdata/test_fetch_listeners.go:2489]: ditto and for the rest

**Review by TAOXUY — APPROVED** [review]

File: src/go/configgenerator/route_generator_test.go (hunk 3 above)
Comment (shuoyang2016) [reply src/go/configgenerator/route_generator_test.go:74]: Yeah, I used go/jsonformatter to pretty print the json which uses two space for indent I think.

File: src/go/configmanager/testdata/test_fixed_mode_dynamic_routing.go (hunk 4 above)
Comment (shuoyang2016) [reply src/go/configmanager/testdata/test_fixed_mode_dynamic_routing.go]: Done. I didn't notice this. My intellij's indent showed two spaces.

File: src/go/configmanager/testdata/test_fetch_listeners.go (hunk 5 above)
Comment (shuoyang2016) [reply src/go/configmanager/testdata/test_fetch_listeners.go]: Done.

**Comment by shuoyang2016** [comment]
> This is exactly why I didn't add unit tests to our config generator in google3 beforehand :) I want to find a better way to do testing internally. Something like this: https://ieftimov.com/post/testing-in-go-golden-files/#keeping-our-golden-files-up-to-date

That makes senses. Another difficult I had is there's no per line diff. I have to copy the entire json and pretty print. If we can use cmp.diff, that would make things a bit better too.

**Pull request merged**